pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -63,7 +72,21 @@ func (pl *ImageLocality) Score(ctx context.Context, state *framework.CycleState,
 	}
 	totalNumNodes := len(nodeInfos)
 
-	score := calculatePriority(sumImageScores(nodeInfo, pod.Spec.Containers, totalNumNodes), len(pod.Spec.Containers))
+	c, err := state.Read(preScoreStateKey)
+	if err != nil {
+		return 0, framework.AsStatus(fmt.Errorf("reading %q from cycleState: %w", preScoreStateKey, err))
+	}
+
+	s, ok := c.(*preScoreState)
+	if !ok {
+		return 0, framework.AsStatus(fmt.Errorf("cannot convert saved state to selectorspread.preScoreState"))
+	}
+
+	if len(s.containersPullAlways) == 0 {
+		return 0, nil;
+	}
+
+	score := calculatePriority(sumImageScores(nodeInfo, s.containersPullAlways, totalNumNodes), len(pod.Spec.Containers))
```

Comment:
oh, just realized that we used to only calculate regular containers. However, I think initContainers should also be counted. @alculquicondor wdyt?

Reply:
I guess we could take them into consideration, but that would be a separate issue/PR

Reply:
Considering init containers start in sequence, I think we should only pick the max value.